pkg/config/finder.go

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright 2019 Chef Software, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package config
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	DefaultChefDirectory            = ".chef"
+	DefaultChefWorkstationDirectory = ".chef-workstation"
+	DefaultFileName                 = "config.toml"
+)
+
+// finds the configuration file (default .chef-workstation/config.toml) inside the current
+// directory and recursively, plus inside the $HOME directory
+func FindChefWorkstationConfigFile() (string, error) {
+	return FindConfigFile(DefaultFileName)
+}
+
+// @afiune do we want to find other files like knife.rb and config.rb?
+//func FindKnifeRbFile() (string, error) {
+//return FindConfigFile("knife.rb")
+//}
+//func FindConfigRbFile() (string, error) {
+//return FindConfigFile("config.rb")
+//}
+
+// finds the provided configuration file inside the current
+// directory and recursively, plus inside the $HOME directory
+func FindConfigFile(name string) (string, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return "", errors.Wrap(err, "unable to detect current directory")
+	}
+
+	var previous string
+	for {
+		if cwd == previous {
+			break
+		}
+
+		//debug("searching config in: %s\n", cwd)
+
+		configFile, exists := configFileExistsInsideDefaultDirectories(cwd, name)
+		if exists {
+			return configFile, nil
+		}
+
+		// save the config file as previous, why? we need a way
+		// to stop the for loop from going in an infinite loop
+		previous = cwd
+		// go down the directory tree
+		cwd = filepath.Join(cwd, "..")
+	}
```

Comment:
What about
```
for root := "/"; cwd != root; cwd = filepath.Join(cwd, "..") {
  configFile, exists := configFileExistsInsideDefaultDirectories(cwd, name)
  if exists {
    return configFile, nil
  }
}
```
I just don't know how that will work on Windows - maybe there is some kind of `os.RootDir()`?

Reply:
Thinking about it more - I don't know that we want to recursively search up the directory until we hit root. What about just looking relative to the `cwd`?

Reply:
Yeah, that sounds good, I can change that behavior to only check the current directory and `$HOME`,
then if the config files are not there, we display an error.